[SK] - implemented tag reporting disable functionality based on confi…

sanintel3 opened the pull request:
Disabling tag report generation implemented
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23issuecomment-124963342%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23issuecomment-125099391%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23issuecomment-125101857%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23issuecomment-125104429%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23issuecomment-125171125%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23issuecomment-125467083%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23issuecomment-125867091%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23issuecomment-125873679%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23issuecomment-127920331%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23issuecomment-127941834%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23issuecomment-128800246%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r35490777%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r35490797%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r35490809%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r35507632%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r35507650%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r35507692%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r35508270%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r35508760%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551860%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551915%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551924%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551935%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551946%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551952%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551955%22%2C%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36552102%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23issuecomment-124963342%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Although%20I%20understand%20that%20this%20fix%20will%20allow%20you%20to%20generate%20the%20report%2C%20I%20would%20rather%20fix%20the%20source%20of%20the%20problem%20instead%20of%20providing%20workaround.%22%2C%20%22created_at%22%3A%20%222015-07-26T09%3A19%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%2C%20%7B%22body%22%3A%20%22May%20I%20know%20is%20it%20not%20a%20value%20add%20to%20make%20this%20reporting%20functionality%20as%20optional%20%28with%20default%20is%20to%20allow%20generation%29%20if%20some%20one%20is%20not%20really%20interested%20in%20tag%20reporting%20and%20just%20want%20feature%20report%2C%20would%20it%20not%20make%20much%20sense%20to%20disable%20that%3F%20On%20the%20other%20side%2C%20as%20per%20my%20initial%20understanding%20I%20tried%20to%20re-factor%20the%20tag%20processing%20logic%2C%20but%20few%20of%20the%20tests%20were%20failing%2C%20I%20%20need%20to%20really%20understand%20the%20crux%20behind%20the%20tag%20processing%20logic%22%2C%20%22created_at%22%3A%20%222015-07-27T06%3A13%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3647001%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/sanintel3%22%7D%7D%2C%20%7B%22body%22%3A%20%22It%20is%20valuable%20but%20still%20it%20would%20be%20nice%20to%20be%20able%20to%20fix%20the%20problem%20which%20was%20the%20motivation%20of%20this%20PR%22%2C%20%22created_at%22%3A%20%222015-07-27T06%3A29%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%2C%20%7B%22body%22%3A%20%22So%20are%20we%20in%20a%20position%20to%20accept%20this%20change%20until%20tag%20processing%20logic%20is%20fixed%3F%22%2C%20%22created_at%22%3A%20%222015-07-27T06%3A40%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3647001%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/sanintel3%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20have%20added%20a%20commit%20to%20restructure%20logic%20to%20populate%20tagMap%20data%20structure.%20Could%20you%20please%20look%20into%20it.%20I%20have%20updated%20test%20expectations%20as%20well%20with%20right%20data.%20Only%20bug%20I%20see%20is%20total%20steps%20count%20in%20tag%20report%20%28tagged%20steps%20%2B%20back%20ground%20steps%29%20is%20not%20proper%20at%20the%20moment.%20will%20try%20to%20fix%20it.%20I%20tried%20to%20generate%20the%20report%20by%20including%20tags%20on%20the%20origin%20json%20file%20%28110%20MB%29%20and%20it%27s%20very%20quick%20and%20I%20don%27t%20see%20any%20performance%20issue%20with%20it.%20ReportInformationTest%20has%20been%20updated%20with%20some%20meta%20data%20which%20will%20help%20in%20understanding%20test%20expectations.%20Please%20review%20and%20let%20me%20know%20your%20thoughts%22%2C%20%22created_at%22%3A%20%222015-07-27T11%3A20%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3647001%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/sanintel3%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20reverted%20original%20commit%20which%20disables%20tag%20reporting%20as%20actual%20issue%20is%20fixed%20now.%22%2C%20%22created_at%22%3A%20%222015-07-28T06%3A34%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3647001%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/sanintel3%22%7D%7D%2C%20%7B%22body%22%3A%20%22May%20I%20know%20would%20you%20like%20to%20accept%20this%20pull%20request%20and%20do%20a%20release%3F%20or%20is%20there%20anything%20that%20you%20would%20like%20to%20point%20out%20that%20requires%20a%20fix.%22%2C%20%22created_at%22%3A%20%222015-07-29T07%3A16%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3647001%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/sanintel3%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20will%20review%20it%20soon%5Cn%5CnSent%20from%20my%20iPhone%5Cn%5Cn%3E%20On%2029%20Jul%202015%2C%20at%2008%3A16%2C%20sanintel3%20%3Cnotifications%40github.com%3E%20wrote%3A%5Cn%3E%20%5Cn%3E%20May%20I%20know%20would%20you%20like%20to%20accept%20this%20pull%20request%20and%20do%20a%20release%3F%20or%20is%20there%20anything%20that%20you%20would%20like%20to%20point%20out%20that%20requires%20a%20fix.%5Cn%3E%20%5Cn%3E%20%5Cu2014%5Cn%3E%20Reply%20to%20this%20email%20directly%20or%20view%20it%20on%20GitHub.%5Cn%3E%20%5Cn%22%2C%20%22created_at%22%3A%20%222015-07-29T07%3A53%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/103770%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kingsleyh%22%7D%7D%2C%20%7B%22body%22%3A%20%22kingsleyh%2C%5Cr%5CnMay%20I%20know%20how%20long%20it%20might%20take%20to%20accept/reject%20this%20pull%20request%3F%22%2C%20%22created_at%22%3A%20%222015-08-05T08%3A53%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3647001%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/sanintel3%22%7D%7D%2C%20%7B%22body%22%3A%20%22Hi%5Cn%5CnSorry%20I%20have%20handed%20the%20project%20over%20to%20Damian%20as%20I%20have%20retired%20from%20working%20on%20it%5Cn%5Cn--k%5Cn%5CnSent%20from%20my%20iPhone%5Cn%5Cn%3E%20On%205%20Aug%202015%2C%20at%2009%3A53%2C%20sanintel3%20%3Cnotifications%40github.com%3E%20wrote%3A%5Cn%3E%20%5Cn%3E%20kingsleyh%2C%5Cn%3E%20May%20I%20know%20how%20long%20it%20might%20take%20to%20accept/reject%20this%20pull%20request%3F%5Cn%3E%20%5Cn%3E%20%5Cu2014%5Cn%3E%20Reply%20to%20this%20email%20directly%20or%20view%20it%20on%20GitHub.%5Cn%3E%20%5Cn%22%2C%20%22created_at%22%3A%20%222015-08-05T09%3A54%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/103770%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kingsleyh%22%7D%7D%2C%20%7B%22body%22%3A%20%22Sorry%20for%20delay%20but%20this%20summer%20is%20hot%20and%20busy%21%5Cr%5Cn%5Cr%5CnI%27ve%20added%20some%20remarks%20but%20I%20need%20to%20check%20the%20impact%20of%20this%20change.%20I%20understand%20and%20accept%20motivation%20so%20this%20is%20rather%20question%20about%20default%20behavior.%22%2C%20%22created_at%22%3A%20%222015-08-07T19%3A21%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20d02a2fcdc58e910e85faa29cd596ca3a6c29f8ce%20src/main/java/net/masterthought/cucumber/ReportBuilder.java%20131%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r35490777%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22why%20you%27ve%20deleted%20javadoc%3F%22%2C%20%22created_at%22%3A%20%222015-07-26T09%3A08%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%2C%20%7B%22body%22%3A%20%22As%20public%20constructor%20is%20replaced%20with%20Builder%20which%20has%20more%20self-explanatory%20methods%20to%20build%20this%20object.%20Constructor%20was%20taking%20more%20than%208%20arguments%20which%20was%20not%20really%20good.%22%2C%20%22created_at%22%3A%20%222015-07-27T06%3A07%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3647001%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/sanintel3%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/main/java/net/masterthought/cucumber/ReportBuilder.java%3AL74-96%22%7D%2C%20%22Pull%2022308dc00ad5c9b88ce0fff377bab638f6bf5f51%20src/test/java/net/masterthought/cucumber/TagsTest.java%2054%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551952%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22code%20formatting%20here%2C%20above%20and%20below%20%28missing%20spaces%29%22%2C%20%22created_at%22%3A%20%222015-08-07T19%3A17%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/test/java/net/masterthought/cucumber/TagsTest.java%3AL110-124%22%7D%2C%20%22Pull%2022308dc00ad5c9b88ce0fff377bab638f6bf5f51%20src/main/java/net/masterthought/cucumber/ReportInformation.java%2094%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551860%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Add%20missing%20blank%20spaces.%22%2C%20%22created_at%22%3A%20%222015-08-07T19%3A16%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/main/java/net/masterthought/cucumber/ReportInformation.java%3AL201-275%22%7D%2C%20%22Pull%2022308dc00ad5c9b88ce0fff377bab638f6bf5f51%20src/test/java/net/masterthought/cucumber/TagsTest.java%2057%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551955%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22break%3F%20probably%20unnecessary%20because%20tags%20are%20unique%22%2C%20%22created_at%22%3A%20%222015-08-07T19%3A17%3A21Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/test/java/net/masterthought/cucumber/TagsTest.java%3AL110-124%22%7D%2C%20%22Pull%20d02a2fcdc58e910e85faa29cd596ca3a6c29f8ce%20src/main/java/net/masterthought/cucumber/ReportBuilder.java%20207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r35490797%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20would%20make%20this%20class%20outer%20and%20no%20static%20-%20%20that%20would%20be%20helpful%20to%20maintain%20it%22%2C%20%22created_at%22%3A%20%222015-07-26T09%3A11%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%2C%20%7B%22body%22%3A%20%22If%20it%20make%20sense%20to%20keep%20java%20doc%20on%20Builder%20class%20%28which%20helps%20in%20creating%20ReportBuilder%20instance%20then%20I%20would%20add%20java%20doc%20for%20those%20builder%20methods%22%2C%20%22created_at%22%3A%20%222015-07-27T06%3A08%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3647001%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/sanintel3%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/main/java/net/masterthought/cucumber/ReportBuilder.java%3AL194-305%22%7D%2C%20%22Pull%2022308dc00ad5c9b88ce0fff377bab638f6bf5f51%20src/main/java/net/masterthought/cucumber/TagObject.java%2024%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36552102%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%27m%20not%20sure%20about%20returning%20value%20without%20backgorunds%20-%20by%20default%20I%20would%20expect%20to%20have%20all%20items%20but%20I%20will%20have%20to%20check%20with%20code%20to%20be%20sure%20about%20consequences...%22%2C%20%22created_at%22%3A%20%222015-08-07T19%3A19%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/main/java/net/masterthought/cucumber/TagObject.java%3AL90-139%22%7D%2C%20%22Pull%20d02a2fcdc58e910e85faa29cd596ca3a6c29f8ce%20src/main/java/net/masterthought/cucumber/ReportInformation.java%209%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r35490809%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22try%20to%20avoid%20changing%20order%20of%20the%20imports%20-%20otherwise%20next%20commit%20will%20change%20this%20order%20back%22%2C%20%22created_at%22%3A%20%222015-07-26T09%3A15%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%2C%20%7B%22body%22%3A%20%22May%20I%20know%20is%20there%20any%20significance%20for%20the%20import%20order%3F%20Is%20there%20any%20specific%20reason%20or%20is%20it%20a%20best%20practice%20to%20avoid%20changes%20to%20import%20statements%3F%22%2C%20%22created_at%22%3A%20%222015-07-27T06%3A09%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3647001%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/sanintel3%22%7D%7D%2C%20%7B%22body%22%3A%20%22It%20not%20about%20the%20order%20but%20about%20potential%20conflicts%20that%20may%20happen%20if%20you%20add%20changes%20that%20can%20be%20avoided%20%22%2C%20%22created_at%22%3A%20%222015-07-27T06%3A27%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%2C%20%7B%22body%22%3A%20%22Agreed%22%2C%20%22created_at%22%3A%20%222015-07-27T06%3A39%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3647001%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/sanintel3%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/main/java/net/masterthought/cucumber/ReportInformation.java%3AL1-7%22%7D%2C%20%22Pull%2022308dc00ad5c9b88ce0fff377bab638f6bf5f51%20src/test/java/net/masterthought/cucumber/TagsTest.java%2019%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551946%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22you%20did%20not%20change%20test%20data%20but%20made%20validation%20less%20restrictive%20%28from%20is%20to%20contains%29%20-%20is%20that%20correct%3F%22%2C%20%22created_at%22%3A%20%222015-08-07T19%3A17%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/test/java/net/masterthought/cucumber/TagsTest.java%3AL73-107%22%7D%2C%20%22Pull%2022308dc00ad5c9b88ce0fff377bab638f6bf5f51%20src/main/java/net/masterthought/cucumber/TagObject.java%2063%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551924%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22we%20should%20be%20ready%20to%20use%20diamond%20operator%5Cr%5Cn%5Cr%5Cn%20%20%20%20new%20ArrayList%3C%3E%28%29%3B%22%2C%20%22created_at%22%3A%20%222015-08-07T19%3A17%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/main/java/net/masterthought/cucumber/TagObject.java%3AL147-165%22%7D%2C%20%22Pull%2022308dc00ad5c9b88ce0fff377bab638f6bf5f51%20src/main/java/net/masterthought/cucumber/TagObject.java%2062%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551915%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22how%20about%20getStatusesWithoutBackGrounds%20%3F%22%2C%20%22created_at%22%3A%20%222015-08-07T19%3A16%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/main/java/net/masterthought/cucumber/TagObject.java%3AL147-165%22%7D%2C%20%22Pull%2022308dc00ad5c9b88ce0fff377bab638f6bf5f51%20src/main/java/net/masterthought/cucumber/TagObject.java%2065%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik/cucumber-reporting/pull/124%23discussion_r36551935%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Make%20double%20check%20whether%20you%20need%20additional%20checking%20such%20as%5Cr%5Cn%5Cr%5Cn%20%20%20%20scenarioTag.hasSteps%28%29%5Cr%5Cn%5Cr%5Cn%5Cr%5Cnbecause%20I%20believe%20that%20if%20there%20is%20no%20scenario%20at%20all%5Cr%5Cn%5Cr%5Cn%20%20%20%20scenarioTag.getScenario%28%29.getSteps%28%29%5Cr%5Cn%5Cr%5Cnsimple%20returns%20empty%20list%20and%20the%20loop%20will%20be%20executed%200%20times%22%2C%20%22created_at%22%3A%20%222015-08-07T19%3A17%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/9612911%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/damianszczepanik%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/main/java/net/masterthought/cucumber/TagObject.java%3AL147-165%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'><img src='http://www.codereviewhub.com/site/github-bar.png' height=40></a>
- [ ] <a href='#crh-comment-Pull d02a2fcdc58e910e85faa29cd596ca3a6c29f8ce src/main/java/net/masterthought/cucumber/ReportBuilder.java 131'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/damianszczepanik/cucumber-reporting/pull/124#discussion_r35490777'>File: src/main/java/net/masterthought/cucumber/ReportBuilder.java:L74-96</a></b>
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> why you've deleted javadoc?
- <a href='https://github.com/sanintel3'><img border=0 src='https://avatars.githubusercontent.com/u/3647001?v=3' height=16 width=16'></a> As public constructor is replaced with Builder which has more self-explanatory methods to build this object. Constructor was taking more than 8 arguments which was not really good.
- [ ] <a href='#crh-comment-Pull d02a2fcdc58e910e85faa29cd596ca3a6c29f8ce src/main/java/net/masterthought/cucumber/ReportBuilder.java 207'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/damianszczepanik/cucumber-reporting/pull/124#discussion_r35490797'>File: src/main/java/net/masterthought/cucumber/ReportBuilder.java:L194-305</a></b>
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> I would make this class outer and no static -  that would be helpful to maintain it
- <a href='https://github.com/sanintel3'><img border=0 src='https://avatars.githubusercontent.com/u/3647001?v=3' height=16 width=16'></a> If it make sense to keep java doc on Builder class (which helps in creating ReportBuilder instance then I would add java doc for those builder methods
- [ ] <a href='#crh-comment-Pull d02a2fcdc58e910e85faa29cd596ca3a6c29f8ce src/main/java/net/masterthought/cucumber/ReportInformation.java 9'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/damianszczepanik/cucumber-reporting/pull/124#discussion_r35490809'>File: src/main/java/net/masterthought/cucumber/ReportInformation.java:L1-7</a></b>
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> try to avoid changing order of the imports - otherwise next commit will change this order back
- <a href='https://github.com/sanintel3'><img border=0 src='https://avatars.githubusercontent.com/u/3647001?v=3' height=16 width=16'></a> May I know is there any significance for the import order? Is there any specific reason or is it a best practice to avoid changes to import statements?
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> It not about the order but about potential conflicts that may happen if you add changes that can be avoided
- <a href='https://github.com/sanintel3'><img border=0 src='https://avatars.githubusercontent.com/u/3647001?v=3' height=16 width=16'></a> Agreed
- [ ] <a href='#crh-comment-General Comment'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/damianszczepanik/cucumber-reporting/pull/124#issuecomment-124963342'>General Comment</a></b>
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> Although I understand that this fix will allow you to generate the report, I would rather fix the source of the problem instead of providing workaround.
- <a href='https://github.com/sanintel3'><img border=0 src='https://avatars.githubusercontent.com/u/3647001?v=3' height=16 width=16'></a> May I know is it not a value add to make this reporting functionality as optional (with default is to allow generation) if some one is not really interested in tag reporting and just want feature report, would it not make much sense to disable that? On the other side, as per my initial understanding I tried to re-factor the tag processing logic, but few of the tests were failing, I  need to really understand the crux behind the tag processing logic
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> It is valuable but still it would be nice to be able to fix the problem which was the motivation of this PR
- <a href='https://github.com/sanintel3'><img border=0 src='https://avatars.githubusercontent.com/u/3647001?v=3' height=16 width=16'></a> So are we in a position to accept this change until tag processing logic is fixed?
- <a href='https://github.com/sanintel3'><img border=0 src='https://avatars.githubusercontent.com/u/3647001?v=3' height=16 width=16'></a> I have added a commit to restructure logic to populate tagMap data structure. Could you please look into it. I have updated test expectations as well with right data. Only bug I see is total steps count in tag report (tagged steps + back ground steps) is not proper at the moment. will try to fix it. I tried to generate the report by including tags on the origin json file (110 MB) and it's very quick and I don't see any performance issue with it. ReportInformationTest has been updated with some meta data which will help in understanding test expectations. Please review and let me know your thoughts
- <a href='https://github.com/sanintel3'><img border=0 src='https://avatars.githubusercontent.com/u/3647001?v=3' height=16 width=16'></a> I reverted original commit which disables tag reporting as actual issue is fixed now.
- <a href='https://github.com/sanintel3'><img border=0 src='https://avatars.githubusercontent.com/u/3647001?v=3' height=16 width=16'></a> May I know would you like to accept this pull request and do a release? or is there anything that you would like to point out that requires a fix.
- <a href='https://github.com/kingsleyh'><img border=0 src='https://avatars.githubusercontent.com/u/103770?v=3' height=16 width=16'></a> I will review it soon
Sent from my iPhone
> On 29 Jul 2015, at 08:16, sanintel3 <notifications@github.com> wrote:
>
> May I know would you like to accept this pull request and do a release? or is there anything that you would like to point out that requires a fix.
>
> —
> Reply to this email directly or view it on GitHub.
>
- <a href='https://github.com/sanintel3'><img border=0 src='https://avatars.githubusercontent.com/u/3647001?v=3' height=16 width=16'></a> kingsleyh,
May I know how long it might take to accept/reject this pull request?
- <a href='https://github.com/kingsleyh'><img border=0 src='https://avatars.githubusercontent.com/u/103770?v=3' height=16 width=16'></a> Hi
Sorry I have handed the project over to Damian as I have retired from working on it
--k
Sent from my iPhone
> On 5 Aug 2015, at 09:53, sanintel3 <notifications@github.com> wrote:
>
> kingsleyh,
> May I know how long it might take to accept/reject this pull request?
>
> —
> Reply to this email directly or view it on GitHub.
>
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> Sorry for delay but this summer is hot and busy!
I've added some remarks but I need to check the impact of this change. I understand and accept motivation so this is rather question about default behavior.
- [ ] <a href='#crh-comment-Pull 22308dc00ad5c9b88ce0fff377bab638f6bf5f51 src/main/java/net/masterthought/cucumber/ReportInformation.java 94'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/damianszczepanik/cucumber-reporting/pull/124#discussion_r36551860'>File: src/main/java/net/masterthought/cucumber/ReportInformation.java:L201-275</a></b>
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> Add missing blank spaces.
- [ ] <a href='#crh-comment-Pull 22308dc00ad5c9b88ce0fff377bab638f6bf5f51 src/main/java/net/masterthought/cucumber/TagObject.java 62'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/damianszczepanik/cucumber-reporting/pull/124#discussion_r36551915'>File: src/main/java/net/masterthought/cucumber/TagObject.java:L147-165</a></b>
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> how about getStatusesWithoutBackGrounds ?
- [ ] <a href='#crh-comment-Pull 22308dc00ad5c9b88ce0fff377bab638f6bf5f51 src/main/java/net/masterthought/cucumber/TagObject.java 63'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/damianszczepanik/cucumber-reporting/pull/124#discussion_r36551924'>File: src/main/java/net/masterthought/cucumber/TagObject.java:L147-165</a></b>
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> we should be ready to use diamond operator
new ArrayList<>();
- [ ] <a href='#crh-comment-Pull 22308dc00ad5c9b88ce0fff377bab638f6bf5f51 src/main/java/net/masterthought/cucumber/TagObject.java 65'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/damianszczepanik/cucumber-reporting/pull/124#discussion_r36551935'>File: src/main/java/net/masterthought/cucumber/TagObject.java:L147-165</a></b>
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> Make double check whether you need additional checking such as
scenarioTag.hasSteps()
because I believe that if there is no scenario at all
scenarioTag.getScenario().getSteps()
simple returns empty list and the loop will be executed 0 times
- [ ] <a href='#crh-comment-Pull 22308dc00ad5c9b88ce0fff377bab638f6bf5f51 src/test/java/net/masterthought/cucumber/TagsTest.java 19'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/damianszczepanik/cucumber-reporting/pull/124#discussion_r36551946'>File: src/test/java/net/masterthought/cucumber/TagsTest.java:L73-107</a></b>
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> you did not change test data but made validation less restrictive (from is to contains) - is that correct?
- [ ] <a href='#crh-comment-Pull 22308dc00ad5c9b88ce0fff377bab638f6bf5f51 src/test/java/net/masterthought/cucumber/TagsTest.java 54'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/damianszczepanik/cucumber-reporting/pull/124#discussion_r36551952'>File: src/test/java/net/masterthought/cucumber/TagsTest.java:L110-124</a></b>
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> code formatting here, above and below (missing spaces)
- [ ] <a href='#crh-comment-Pull 22308dc00ad5c9b88ce0fff377bab638f6bf5f51 src/test/java/net/masterthought/cucumber/TagsTest.java 57'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/damianszczepanik/cucumber-reporting/pull/124#discussion_r36551955'>File: src/test/java/net/masterthought/cucumber/TagsTest.java:L110-124</a></b>
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> break? probably unnecessary because tags are unique
- [ ] <a href='#crh-comment-Pull 22308dc00ad5c9b88ce0fff377bab638f6bf5f51 src/main/java/net/masterthought/cucumber/TagObject.java 24'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/damianszczepanik/cucumber-reporting/pull/124#discussion_r36552102'>File: src/main/java/net/masterthought/cucumber/TagObject.java:L90-139</a></b>
- <a href='https://github.com/damianszczepanik'><img border=0 src='https://avatars.githubusercontent.com/u/9612911?v=3' height=16 width=16'></a> I'm not sure about returning value without backgorunds - by default I would expect to have all items but I will have to check with code to be sure about consequences...


<a href='https://www.codereviewhub.com/damianszczepanik/cucumber-reporting/pull/124?mark_as_completed=1'><img src='http://www.codereviewhub.com/site/github-mark-as-completed.png' height=26></a>&nbsp;<a href='https://www.codereviewhub.com/damianszczepanik/cucumber-reporting/pull/124?approve=1'><img src='http://www.codereviewhub.com/site/github-approve.png' height=26></a>&nbsp;<a href='https://github.com/damianszczepanik/cucumber-reporting/pull/124'><img src='http://www.codereviewhub.com/site/github-refresh.png' height=26></a>
<a href='#crh-end'></a>

damianszczepanik commented on src/main/java/net/masterthought/cucumber/ReportBuilder.java:
```diff
@@ -117,53 +74,22 @@ public boolean isFlashCharts() {
+        return new ReportBuilder.Builder();
     }
 
-    /**
```
why you've deleted javadoc?

sanintel3 replied:
As public constructor is replaced with Builder which has more self-explanatory methods to build this object. Constructor was taking more than 8 arguments which was not really good.

damianszczepanik commented on src/main/java/net/masterthought/cucumber/ReportBuilder.java:
```diff
@@ -264,4 +194,111 @@ private void generateErrorPage(Exception exception) throws IOException {
         System.out.println(exception);
             }
+
+    public static final class Builder {
```
I would make this class outer and no static -  that would be helpful to maintain it

sanintel3 replied:
If it make sense to keep java doc on Builder class (which helps in creating ReportBuilder instance then I would add java doc for those builder methods

damianszczepanik commented on src/main/java/net/masterthought/cucumber/ReportInformation.java:
```diff
@@ -1,11 +1,6 @@
-import java.util.List;
-import java.util.Map;
-
+import com.googlecode.totallylazy.Sequence;
```
try to avoid changing order of the imports - otherwise next commit will change this order back

sanintel3 replied:
May I know is there any significance for the import order? Is there any specific reason or is it a best practice to avoid changes to import statements?

damianszczepanik replied:
It not about the order but about potential conflicts that may happen if you add changes that can be avoided

sanintel3 replied:
Agree.

damianszczepanik commented:
Although I understand that this fix will allow you to generate the report, I would rather fix the source of the problem instead of providing workaround.

sanintel3 commented:
May I know is it not a value add to make this reporting functionality as optional (with default is to allow generation) if some one is not really interested in tag reporting and just want feature report, would it not make much sense to disable that? On the other side, as per my initial understanding I tried to re-factor the tag processing logic, but few of the tests were failing, I  need to really understand the crux behind the tag processing logic

damianszczepanik commented:
It is valuable but still it would be nice to be able to fix the problem which was the motivation of this PR

sanintel3 commented:
So are we in a position to accept this change until tag processing logic is fixed?

sanintel3 commented:
I have added a commit to restructure logic to populate tagMap data structure. Could you please look into it. I have updated test expectations as well with right data. Only bug I see is total steps count in tag report (tagged steps + back ground steps) is not proper at the moment. will try to fix it. I tried to generate the report by including tags on the origin json file (110 MB) and it's very quick and I don't see any performance issue with it. ReportInformationTest has been updated with some meta data which will help in understanding test expectations. Please review and let me know your thoughts

sanintel3 commented:
I reverted original commit which disables tag reporting as actual issue is fixed now.

sanintel3 commented:
May I know would you like to accept this pull request and do a release? or is there anything that you would like to point out that requires a fix.

kingsleyh commented:
I will review it soon

Sent from my iPhone

> On 29 Jul 2015, at 08:16, sanintel3 <notifications@github.com> wrote:
> 
> May I know would you like to accept this pull request and do a release? or is there anything that you would like to point out that requires a fix.
> 
> —
> Reply to this email directly or view it on GitHub.
>

sanintel3 commented:
kingsleyh,
May I know how long it might take to accept/reject this pull request?

kingsleyh commented:
Hi

Sorry I have handed the project over to Damian as I have retired from working on it

--k

Sent from my iPhone

> On 5 Aug 2015, at 09:53, sanintel3 <notifications@github.com> wrote:
> 
> kingsleyh,
> May I know how long it might take to accept/reject this pull request?
> 
> —
> Reply to this email directly or view it on GitHub.
>

damianszczepanik commented on src/main/java/net/masterthought/cucumber/ReportInformation.java:
```diff
@@ -207,66 +201,74 @@ private void processTags() {
+            Sequence<String> featureLevelTags = feature.getTagList();
 
-                }
+            if(Util.itemExists(scenarios)){
```
Add missing blank spaces.

damianszczepanik commented on src/main/java/net/masterthought/cucumber/TagObject.java:
```diff
@@ -143,6 +147,18 @@ public int getNumberOfPending() {
         return statuses;
     }
 
+    private List<Status> getStatusesByExcludingBackGround() {
```
how about getStatusesWithoutBackGrounds ?

damianszczepanik commented on src/main/java/net/masterthought/cucumber/TagObject.java:
```diff
@@ -143,6 +147,18 @@ public int getNumberOfPending() {
     }
 
+    private List<Status> getStatusesByExcludingBackGround() {
+        List<Status> statuses = new ArrayList<Status>();
```
we should be ready to use diamond operator

    new ArrayList<>();

damianszczepanik commented on src/main/java/net/masterthought/cucumber/TagObject.java:
```diff
@@ -143,6 +147,18 @@ public int getNumberOfPending() {
+    private List<Status> getStatusesByExcludingBackGround() {
+        List<Status> statuses = new ArrayList<Status>();
+        for (ScenarioTag scenarioTag : scenarios) {
+            if (!scenarioTag.getScenario().isBackground() && scenarioTag.hasSteps()) {
```
Make double check whether you need additional checking such as

    scenarioTag.hasSteps()


because I believe that if there is no scenario at all

    scenarioTag.getScenario().getSteps()

simple returns empty list and the loop will be executed 0 times

damianszczepanik commented on src/test/java/net/masterthought/cucumber/TagsTest.java:
```diff
@@ -72,19 +73,34 @@ public void shouldgetTotalTagScenariosFailed() {
+            tagNames.add(tagObject.getTagName());
+        }
+
+        assertTrue(tagNames.contains("@tag1"));
```
you did not change test data but made validation less restrictive (from is to contains) - is that correct?

damianszczepanik commented on src/test/java/net/masterthought/cucumber/TagsTest.java:
```diff
@@ -94,7 +110,14 @@ public void shouldGetTagScenariosForTag1() {
     public void shouldGetTagScenariosForTag2() {
-        List<ScenarioTag> scenarios = reportInformation.getTags().get(1).getScenarios();
+        TagObject tagObject = null;
+        for(TagObject temp : reportInformation.getTags()){
```
code formatting here, above and below (missing spaces)

damianszczepanik commented on src/test/java/net/masterthought/cucumber/TagsTest.java:
```diff
@@ -94,7 +110,14 @@ public void shouldGetTagScenariosForTag1() {
+        for(TagObject temp : reportInformation.getTags()){
+            if(temp.getTagName().equals("@tag2")){
+                tagObject = temp;
+            }
```
break? probably unnecessary because tags are unique

damianszczepanik commented on src/main/java/net/masterthought/cucumber/TagObject.java:
```diff
@@ -90,45 +90,49 @@ public String getDurationOfSteps() {
     /** No-parameters method required for velocity template. */
     public int getNumberOfPasses() {
-        return getNumberOfStatus(Status.PASSED);
+        return getNumberOfStatusExcludingBackGround(Status.PASSED);
```
I'm not sure about returning value without backgorunds - by default I would expect to have all items but I will have to check with code to be sure about consequences...

damianszczepanik commented:
Sorry for delay but this summer is hot and busy!

I've added some remarks but I need to check the impact of this change. I understand and accept motivation so this is rather question about default behavior.

damianszczepanik commented:
Plugin has been refactored recently so tag generation works pretty fast now.
